src/app/lib/access/firebase/firebase_repository.dart

```diff
@@ -0,0 +1,101 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:app/access/forced_update/data/version.dart';
+import 'package:app/access/forced_update/minimum_version_repository.dart';
+import 'package:app/access/kill_switch/kill_switch_repository.dart';
+import 'package:app/firebase_options.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_remote_config/firebase_remote_config.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:get_it/get_it.dart';
+import 'package:logger/logger.dart';
+import 'package:rxdart/rxdart.dart';
+
+/// Repository that gets the values from Firebase Remote Config.
+/// Implements the [MinimumVersionRepository] and [KillSwitchRepository].
+abstract interface class FirebaseRemoteConfigRepository
```

Comment:
Why do we have Firebase in the name of the abstraction?

Reply:
Should I remove the interface part? at the beginning, I wanted to do something else, but it ended up just being the implementation of the other repositories

Reply:
What is what? What are the dependencies between each piece?
Remember that data sources need to be abstract because they need to be mocked in functional tests.

Reply:
I dont think we have to test / mock this. This is just a straight implementation of those repositories.

Reply:
Yep it's clearer now, thanks!